pkg/cli/start.go

```diff
@@ -118,9 +119,53 @@ func initInsecure() error {
 	return nil
 }
 
+// maxSizePerProfile is the maximum total size in bytes for profiles per
+// profile type.
+var maxSizePerProfile = envutil.EnvOrDefaultInt64(
```

Comment:
That goroutine profiles can get quite big sounds like an argument for limiting by size.